hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/ReadMapOrCacheP.java

```diff
@@ -99,11 +99,13 @@
  */
 public final class ReadMapOrCacheP<F extends CompletableFuture, B, R> extends AbstractProcessor {
 
-    private static final int MAX_FETCH_SIZE = 16384;
+    private static final int MAX_FETCH_SIZE = 2048;
+    private static final int MAX_PARALLEL_READ = 5;
```

Comment:
Did you do any tests to measure these values don't deteriorate the throughput? I'm not sure about them, maybe 5 is too low, maybe not. Local parallelism is 1 by default, one can increase the number of parallel reads by increasing the local parallelism. But we should do some basic benchmarks and put the results here to a comment to this value.

Reply:
I've run some benchmarks on my local, 6 cores 32GB :

Environment:
2 Nodes (separate JVM), 5G each, map populated with 10M items (integer)

Pipeline:
```
        p.readFrom(Sources.map("map"))
                .writeTo(Sinks.noop());
```

First I run 20 jobs for warmup
For each value of the parameter (MAX_PARALLEL_READ) I run 100 jobs and get the average

Results:

Before changes ->  average: 860

Since the partition count is 271, and we have 2 nodes, each processor will have around 135 partitions
setting MAX_PARALLEL_READ above 135 is effectively same so starting the testing from 160

MAX_FETCH_SIZE=16384, MAX_PARALLEL_READ=160-> average: 857
MAX_FETCH_SIZE=16384, MAX_PARALLEL_READ=80  -> average: 859
MAX_FETCH_SIZE=16384, MAX_PARALLEL_READ=40  -> average: 800
MAX_FETCH_SIZE=16384, MAX_PARALLEL_READ=20  -> average: 790
MAX_FETCH_SIZE=16384, MAX_PARALLEL_READ=10   -> average: 771
MAX_FETCH_SIZE=16384, MAX_PARALLEL_READ=5    -> average: 779
MAX_FETCH_SIZE=16384, MAX_PARALLEL_READ=3    -> average: 761
MAX_FETCH_SIZE=16384, MAX_PARALLEL_READ=1     -> average: 1106

MAX_FETCH_SIZE=2048, MAX_PARALLEL_READ=160-> average: 810
MAX_FETCH_SIZE=2048, MAX_PARALLEL_READ=80  -> average: 776
MAX_FETCH_SIZE=2048, MAX_PARALLEL_READ=40  -> average: 786
MAX_FETCH_SIZE=2048, MAX_PARALLEL_READ=20  -> average: 775
MAX_FETCH_SIZE=2048, MAX_PARALLEL_READ=10   -> average: 757
MAX_FETCH_SIZE=2048, MAX_PARALLEL_READ=5    -> average: 720
MAX_FETCH_SIZE=2048, MAX_PARALLEL_READ=3    -> average: 688
MAX_FETCH_SIZE=2048, MAX_PARALLEL_READ=1     -> average: 1191

Although `MAX_PARALLEL_READ=3` has better results, I believe it is related to the core count in the computer.
And a single node run showed very close results for `3 -> 1059`  `5 -> 1061`.


Looks like by introducing this limit to the parallel read, we are actually improving the throughput.

Reply:
Add a link to this comment to the constants.